### PR TITLE
feat(esl-media-query): extend `ESLMediaRuleList.parseTuple` with ability to pass Parser

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

**Review by julia-murashko — APPROVED** [review]

File: src/modules/esl-media-query/README.md [hunk 1]
```diff
@@ -125,12 +125,27 @@ It implements an observable interface, so you can observe active value changes.
 ESLMediaRuleList can be parsed from `<default_value> | <ESL Media Rule> | <ESL Media Rule>` string.
 
 ESLMediaRuleList should be generalized with the result value type and according payload parser.
-- `ESLMediaRuleList.STRING_PARSER` - out of the box string payload parser.
+- `ESLMediaRuleList.STRING_PARSER` - out of the box string payload parser. (you can use String constructor instead)
```
Comment (abarmina) [inline src/modules/esl-media-query/README.md:128]: cosmetic:
```suggestion
- `ESLMediaRuleList.STRING_PARSER` - out of the box string payload parser (you can use String constructor instead).
```

File: src/modules/esl-media-query/README.md [hunk 2]
```diff
@@ -125,12 +125,27 @@ It implements an observable interface, so you can observe active value changes.
...
 
 - ` @XS => {option: 1} | @+SM => {option: 2}` - the RuleList result depends on environment and equals `{option: 1}` 
   on XS breakpoint, and equals `{option: 2}` on SM and upper breakpoints.
+
+Parsing queries examples:
+
+```typescript
+// Basic parsing
+ESLMediaRuleList.parse('1 | @XS => 2'); // first query from the sample above
+ESLMediaRuleList.parse('1 | @XS => 2', String); // the same as sumple above
```
[the remark this answers is not in the document]
Comment (abarmina) [reply src/modules/esl-media-query/README.md]:
```suggestion
ESLMediaRuleList.parse('1 | @XS => 2', String); // the same as the sample above
```

File: src/modules/esl-media-query/README.md [hunk 3]
```diff
@@ -125,12 +125,27 @@ It implements an observable interface, so you can observe active value changes.
...
+ESLMediaRuleList.parse('1 | @XS => 2', Number); // first query from the sample above that store numeric values
+ESLMediaRuleList.parse('@XS => {option: 1} | @+SM => {option: 2}', ESLMediaRuleList.OBJECT_PARSER); // second query from the sample above with an object payloads
+ESLMediaRuleList.parse('@XS => {option: 1} | @+SM => {option: 2}', evaluate); // the same as sample above
```
Comment (abarmina) [inline src/modules/esl-media-query/README.md:146]:
```suggestion
ESLMediaRuleList.parse('@XS => {option: 1} | @+SM => {option: 2}', evaluate); // the same as the sample above 
```

File: src/modules/esl-media-query/core/esl-media-rule-list.ts [hunk 4]
```diff
@@ -93,7 +102,7 @@ export class ESLMediaRuleList<T = any> extends Observable<RuleChangedCallback<T>
     this._rules.forEach((rule) => rule.removeListener(this._onMatchChanged));
   }
 
-  /** List of inner {@link ESLMediaRule}s */
+  /** Array of {@link ESLMediaRule}s that forms current {@link ESLMediaRuleList} */
```
Comment (abarmina) [inline src/modules/esl-media-query/core/esl-media-rule-list.ts:105]:
```suggestion
  /** Array of {@link ESLMediaRule}s that forms the current {@link ESLMediaRuleList} */
```

**Review by abarmina — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Pull request merged**